java-frontend/src/main/java/org/sonar/java/regex/JavaUnicodeEscapeParser.java

```diff
@@ -62,7 +63,13 @@ public void moveNext() {
     }
     int startIndex = index;
     char ch;
-    if (index < sourceText.length() - 1 && sourceText.charAt(index) == '\\' && sourceText.charAt(index + 1) == 'u') {
+    if (sourceText.charAt(index) == '\\') {
+      isEscaping = !isEscaping;
```

Comment:
It looks to me as if something like `\t\u1234` isn't going to be recognized as a Unicode escape this way because the `\t` sets `isEscaping` to `true` and the backslash from `\u` sets it to false.